### PR TITLE
feat(TextFieldBase): expose internal TextFieldBase component for advanced usage

### PR DESCRIPTION
**Pull request by sergivillar**

### CODE REVIEW

File: src/text-field-base.tsx [hunk 1]
```diff
@@ -565,9 +565,9 @@ const TextFieldBase = React.forwardRef<any, TextFieldBaseProps>(
...
             </React.Suspense>
         ) : (
-            <TextFieldBaseComponent {...props} id={id} ref={ref} />
+            <TextFieldBase {...props} id={id} ref={ref} />
         );
     }
 );
 
-export default TextFieldBase;
+export default TextFieldBaseAutosuggest;
```
Comment (atabel) [inline src/text-field-base.tsx:573]: it's weird that the file is called text-field-base and the default export in this file is not the TextFieldBase, but there is also a named export with that name :/

Comment (sergivillar) [reply src/text-field-base.tsx:573]: totally true, I think the same, but.... any better naming? for me it make sense the names in the file, I could rename the file to 'text-field-base-autosuggest' but for me that's a bit weird too....

Comment (pladaria) [reply src/text-field-base.tsx:573]: I'd make both a named export (no default)

Comment (sergivillar) [reply src/text-field-base.tsx]: changed, take a look

**Review by atabel — APPROVED** [review]

**Pull request merged**